content/5_Plat/Sqrt.mdx

```diff
@@ -50,58 +50,149 @@ $$
+$i$-th block represents the sum of the elements from the range $[i\cdot
+\texttt{blocksz},(i + 1)\cdot \texttt{blocksz})$, and $R=\left\lfloor
+\frac{r}{\texttt{blocksz}} \right\rfloor$.
 
 Finally, $\sum_{i=l}^{r} \texttt{nums}[i]$ is the difference between the two
-sums $\sum_{i=0}^{r}\texttt{nums}[i]$ and $\sum_{i=0}^{l-1}\texttt{nums}[i]$, which each are calculated in $\mathcal{O}(\sqrt N)$.
+sums $\sum_{i=0}^{r}\texttt{nums}[i]$ and $\sum_{i=0}^{l-1}\texttt{nums}[i]$,
+which each are calculated in $\mathcal{O}(\sqrt N)$.
 
 <LanguageSection>
 
 <CPPSection>
 
 ```cpp
-//CodeSnip{CPP Short Template}
+#include <bits/stdc++.h>
+using namespace std;
 
-ll blocks[450]; // 450^2 > 2e5
-int nums[(int)(2e5 + 1)];
-int block_sz;
+vector<long long> blocks;
+vector<int> nums;
+int block_size;
 
-void upd(int x, int v) { // O(1) update
-	blocks[x / block_sz] -= nums[x];
+/** O(1) update to set nums[x] to v */
+void update(int x, int v) { 
+	blocks[x / block_size] -= nums[x];
 	nums[x] = v;
-	blocks[x / block_sz] += nums[x];
+	blocks[x / block_size] += nums[x];
 }
 
-ll query(int r) {
-	ll ret = 0;
-	for(int i = 0; i < r / block_sz; i++) ret += blocks[i];
-	for(int i = r / block_sz * block_sz; i < r; i++) ret += nums[i];
-	return ret;
+/** O(sqrt(n)) query for sum of [0, r) */
+long long query(int r) {
+	long long res = 0;
+	for (int i = 0; i < r / block_size; i++) {
+		res += blocks[i];
+	}
+	for (int i = (r / block_size) * block_size; i < r; i++) {
+		res += nums[i];
+	}
+	return res;
 }
 
-ll query(int l, int r) {
+/** O(sqrt(n)) query for sum of [l, r) */
+long long query(int l, int r) {
 	return query(r) - query(l - 1);
 }
 
 int main() {
-	setIO();
+	int n, q; 
+	cin >> n >> q;
+	block_size = (int) ceil(sqrt(n));
+	blocks.assign(block_size, 0);
+	nums.assign(n, 0);
+
+	for (int i = 0; i < n; i++) {
+		cin >> nums[i];
+	}
+	for (int i = 0; i < n; i++) {
+		blocks[i / block_size] += nums[i];
+	}
+	for (int i = 0; i < q; i++) {
+		int t, l, r; 
+		cin >> t >> l >> r;
+		if (t == 1) {
+			update(l - 1, r);
+		} else {
+			cout << query(l, r) << "\n";
+		}
+	}
+}
+```
+
+</CPPSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
 
-	int n, q; cin >> n >> q;
-	block_sz = (int) sqrt(n);
+public class DRSQ {
+	static int blockSize;
+	static long[] blocks;
+	static int[] nums;
 
-	for(int i = 0; i < n; ++i) cin >> nums[i];
-	for(int i = 0; i < n; ++i) blocks[i / block_sz] += nums[i];
-	for(int i = 0; i < q; ++i) {
-		int t, l, r; cin >> t >> l >> r;
-		if (t == 1) upd(l - 1, r);
-		else cout << query(l, r) << "\n";
+	public static void main(String[] args) throws IOException {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		PrintWriter pw = new PrintWriter(System.out);
+		StringTokenizer st = new StringTokenizer(br.readLine());
+
+		int n = Integer.parseInt(st.nextToken());
+		int q = Integer.parseInt(st.nextToken());
+		blockSize = (int) Math.ceil(Math.sqrt(n));
```

Comment:
Try to make the code between languages match up as much as possible (including variables)

Reply:
Should I follow the code convention for java or consisten-fy the variable names? They are essentially the same name except the case

Reply:
It doesn't really matter, just make sure that the explanation matches with the code